src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -103,14 +103,14 @@ internal async ValueTask<TValue> InvokeAsync<TValue>(long targetInstanceId, stri
             return await InvokeAsync<TValue>(targetInstanceId, identifier, CancellationToken.None, args);
         }
 
-        internal ValueTask<TValue> InvokeAsync<TValue>(
+        internal ValueTask<TValue?> InvokeAsync<TValue>(
             long targetInstanceId,
             string identifier,
             CancellationToken cancellationToken,
             object?[]? args)
         {
             var taskId = Interlocked.Increment(ref _nextPendingTaskId);
-            var tcs = new TaskCompletionSource<TValue>(TaskContinuationOptions.RunContinuationsAsynchronously);
+            var tcs = new TaskCompletionSource<TValue?>(TaskContinuationOptions.RunContinuationsAsynchronously);
             if (cancellationToken != default)
```

Comment:
Not part of this pull request, but this has a form that is much easier to read:

```suggestion
            if (cancellationToken.CanBeCancelled)
```